lib/velum/dex/ldap.rb

```diff
@@ -0,0 +1,74 @@
+require "base64"
+
+module Velum
+  # This class offers the integration between ruby and the Saltstack API.
+  module Dex
+    class << self
+      def ldap_connectors_as_pillar
+        ldap_connectors = []
+        begin
+          ldap_connectors = DexConnectorLdap.all.map do |con|
+            {
+              "type"            => "ldap",
+              "id"              => con.id,
+              "name"            => con.name,
+
+              # Combine host and port since they ultimately
+              #   feed into a single line of config for dex
+              "server"          => "#{con.host}:#{con.port}",
+              "start_tls"       => con.start_tls,
+              "root_ca_data"    => Base64.encode64(con.certificate.try(:certificate) || ""),
+              "bind"            => generate_bind_block(con), # Place basic bind information together
+              "user"            => generate_user_block(con), # Place user stuff together
+              "group"           => generate_group_block(con), # Place group stuff together
+              "username_prompt" => con.username_prompt
+            }
+          end
+        rescue ActiveRecord::StatementInvalid => e
+          # Silently ignore non existence of table so that cluster can start without it
+          # :nocov:
+          raise e unless e.message.include? "dex_connectors_ldap' doesn't exist"
+          # :nocov:
+        end
+        ldap_connectors
```

Comment:
you could save a few lines of code and dont need to cache the connectors like this:

```ruby
def ldap_connectors_as_pillar
  DexConnectorLdap.all.map do |con|
￼    {
￼       ...
    }
  end
rescue ActiveRecord::StatementInvalid => e
  ...
end
```

but i wonder do we need to have the `rescue` block at all? or is it still a remainder of your issue that the table wasn't create automatically? it's also only catching a very specific exception, which normally is good, but i think the table should always be there, so it's not really needed

Reply:
Without this rescue the code fails CI. The table is not created automatically. I don't know why. Once it is known why the table isn't created, and how to make it so that it is, this rescue can be dropped.

I added this rescue primarily to prove that CI was failing specifically because of this, rather than only speculating about it.

Using a rescue within a function without begin is interesting; I agree it would be cleaner. As you state though, this code is entirely unneeded when everything works correctly. It doesn't so it is a hack at the moment.

Reply:
Great, that was my speculation that this was true. ( that everything will be fine once the PR is merged )

How am I supposed to get it to pass CI without a rescue block? If it doesn't pass CI the checks on Github will fail, and then merging will be prevented. This is why I added the rescue block.

Once it is accepted once, an additional PR to remove the rescue block would then be able to succeed, correct?

Reply:
We can either leave it in, and remove in another PR.. or better yet, fix the Dev tooling to apply the PRs schema.rb and re-run the jobs once that's merged.

I'd be fine either way, but obviously, it'll be best to fix the tooling issue :)

Reply:
@MaximilianMeister thoughts? How much effort is fixing the tooling?

(I don't want to block progress on tooling issues for the sake of an excess rescue block!)

Reply:
Alright; I will drop the rescue block once I see the latest rebase and patch addition pass CI again.

Reply:
created https://trello.com/c/p9ZNCCss/1054-velum-devenv-enable-testing-database-changes to track this

Reply:
Am trying to get a successful integration test run of this. It has failed twice so far. It looks random. Am trying a third time.